snapshot/git/gitfiler/pool.go

```diff
@@ -0,0 +1,89 @@
+package gitfiler
+
+import (
+	"github.com/scootdev/scoot/snapshot/git/repo"
+)
+
+// repoAndError holds a repo and an error (just so we can pass it across a channel)
+type repoAndError struct {
+	repo *repo.Repository
+	err  error
+}
+
+// NewRepoPool creates a new RepoPool populated with existing repos and a getter that can get new ones
+func NewRepoPool(getter RepoGetter, repos []*repo.Repository, doneCh chan struct{}) *RepoPool {
+	freeList := make([]repoAndError, len(repos))
+	for i, v := range repos {
+		freeList[i] = repoAndError{repo: v}
+	}
+
+	p := &RepoPool{
+		getter:    getter,
+		releaseCh: make(chan repoAndError),
+		reserveCh: make(chan repoAndError),
+		doneCh:    doneCh,
+		freeList:  freeList,
+	}
+	go p.loop()
+	return p
+}
+
+// RepoPool lets clients Get a Repository for an operation, and then Release it when done
+// RepoPool can create a new Repository by using a supplied RepoGetter.
+// New repos can be added by the client by Release'ing a new Repository
+// Cf. sync's Pool
+type RepoPool struct {
```

Comment:
Can we instead get away with a pool without loop() that calls getter.Get() but simply destroys the repo when released? If reference/shared cloning is cheap then removing the directory wholesale might even be faster than running 'git clean' on every checkout. I mean this is a fairly standard pool impl so I may be overthinking this.